File: cs/Iris/ParserUtils/LexParseException/IncompatibleTypesException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.ParserUtils.LexParseException
{
    public class IncompatibleTypesException : BaseException
    {
        public ObjectType expectedType;
        public ObjectType givenType;

        public IncompatibleTypesException(int lNum, int cNum, ObjectType expected, ObjectType given)
            : base(lNum, cNum)
        {
            expectedType = expected;
            givenType = given;
        }

        public override string ExceptionToString()
        {
            StringBuilder str = new StringBuilder();

            str.Append("incompatible types, expected ");
            str.Append(expectedType.name);
            str.Append(" but ");
            str.Append(givenType.name);
            str.Append(" given");

            return str.ToString();
        }
    }
}

File: java/Iris/src/com/rokuan/iris/exception/IncompatibleTypesException.java
package com.rokuan.iris.exception;

import com.rokuan.iris.types.ObjectType;

public class IncompatibleTypesException extends BaseException
{
	public ObjectType expectedType;
	public ObjectType givenType;

	public IncompatibleTypesException(int lNum, int cNum, ObjectType expected, ObjectType given)
	{
		super(lNum, cNum);
		expectedType = expected;
		givenType = given;
	}

	public String ExceptionToString()
	{
		StringBuilder str = new StringBuilder();
		
		str.append(super.stringTrace());

		str.append("incompatible types, expected ");
		str.append(expectedType.name);
		str.append(" but ");
		str.append(givenType.name);
		str.append(" given");

		return str.toString();
	}
}
